sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -299,13 +309,10 @@ class UnboundedOffsetWindowFunctionFrame(
   assert(offset > 0)
 
   override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
-    input = rows
-    if (offset > input.length) {
+    if (offset > rows.length) {
```

Comment:
+1 for @imback82 's suggestion.

Reply:
I was trying to avoid calling `input.generateIterator()`, as it was not called before when `offset > rows.length`, and it may cause perf regression as the name `generateIterator` sounds expensive.

Reply:
ah ok, thanks for the explanation.